front/src/modules/trainschedule/components/DriverTrainScheduleV2/utils.ts

```diff
@@ -87,6 +78,26 @@ export function getActualPositionLeft(givenPosition: number, speed: PositionSpee
   return speed[speedPosition].position / 1000;
 }
 
+// Get the Vmax at a givenPosition (in meters), using vmax (MRSP in m/s)
+// Returns the current Vmax if in the middle of an interval, or
+// the min of the Vmax before and after if exactly at a bound.
+export function getActualVmaxMetersPerSecond(givenPosition: number, vmax: RangeSpeeds) {
+  const vmaxUpperBoundIndex = d3.bisectRight(vmax.internal_boundaries, givenPosition);
+  // Error case: vmax doesn't respect the RangeSpeeds specifications on the lists' lengths
+  if (vmaxUpperBoundIndex > vmax.speeds.length - 1) return 0;
+  // If exactly on a speed-limit change, use the minimal value of both side
+  if (
+    vmaxUpperBoundIndex > 0 &&
+    vmax.internal_boundaries[vmaxUpperBoundIndex - 1] === givenPosition
+  )
+    return Math.min(vmax.speeds[vmaxUpperBoundIndex], vmax.speeds[vmaxUpperBoundIndex - 1]);
+  return vmax.speeds[vmaxUpperBoundIndex];
+}
+
+export function getActualVmax(givenPosition: number, vmax: RangeSpeeds) {
+  return Math.round(getActualVmaxMetersPerSecond(givenPosition, vmax) * 3.6);
```

Comment:
Don't use magic numbers, you can use instead a conversion function, for instance msToKmh
```suggestion
  return Math.round(msToKmh(getActualVmaxMetersPerSecond(givenPosition, vmax)));
```